python/ctsm/site_and_regional/tower_site.py

```diff
@@ -0,0 +1,413 @@
+        Otherwise, adjust user_nl to include different mfilt, nhtfrq, and variables in hist_fincl1.
+        """
+        user_nl_fname = os.path.join(case_root, "user_nl_clm")
+        user_nl_lines = None
+        if run_type == "transient":
+            if self.finidat:
+                user_nl_lines = [
+                    "finidat = '{}/inputdata/lnd/ctsm/initdata/{}'".format(rundir, self.finidat)
+                ]
+        else:
+            user_nl_lines = [
+                "hist_fincl2 = ''",
+                "hist_mfilt = 20",
+                "hist_nhtfrq = -8760",
+                "hist_empty_htapes = .true.",
+            ] + site_lines
+
+        if user_nl_lines:
+            with open(user_nl_fname, "a") as nl_file:
+                for line in user_nl_lines:
+                    nl_file.write("{}\n".format(line))
+
+    def set_ref_case(self, case):
+        """
+        Set an existing case as the reference case, eg for use with spinup.
+        """
+        rundir = case.get_value("RUNDIR")
+        case_root = case.get_value("CASEROOT")
+        if case_root.endswith(".postad"):
+            ref_case_root = case_root.replace(".postad", ".ad")
+            root = ".ad"
+        else:
+            ref_case_root = case_root.replace(".transient", ".postad")
+            root = ".postad"
+        if not os.path.isdir(ref_case_root):
+            logger.warning(
+                "ERROR: spinup must be completed first, could not find directory %s", ref_case_root
+            )
+            return False
+
+        with Case(ref_case_root) as refcase:
+            refrundir = refcase.get_value("RUNDIR")
+        case.set_value("RUN_REFDIR", refrundir)
+        case.set_value("RUN_REFCASE", os.path.basename(ref_case_root))
+        refdate = None
+        for reffile in glob.iglob(refrundir + "/{}{}.clm2.r.*.nc".format(self.name, root)):
+            m_searched = re.search(r"(\d\d\d\d-\d\d-\d\d)-\d\d\d\d\d.nc", reffile)
+            if m_searched:
+                refdate = m_searched.group(1)
+            symlink_force(reffile, os.path.join(rundir, os.path.basename(reffile)))
+        logger.info("Found refdate of %s", refdate)
+        if not refdate:
+            logger.warning("Could not find refcase for %s", case_root)
+            return False
+
+        for rpfile in glob.iglob(refrundir + "/rpointer*"):
+            safe_copy(rpfile, rundir)
+        if not os.path.isdir(os.path.join(rundir, "inputdata")) and os.path.isdir(
+            os.path.join(refrundir, "inputdata")
+        ):
+            symlink_force(os.path.join(refrundir, "inputdata"), os.path.join(rundir, "inputdata"))
+
+        case.set_value("RUN_REFDATE", refdate)
+        if case_root.endswith(".postad"):
+            case.set_value("RUN_STARTDATE", refdate)
+        # NOTE: if start options are set, RUN_STARTDATE should be modified here
+        return True
+
+    # pylint: disable=too-many-statements
```

Comment:
We should try to avoid disabling too-many-statements as that tells us our code is too long and too complex and should be broken up.

Reply:
Add a TODO.